mlflow/tracking/fluent.py

```diff
@@ -146,6 +147,13 @@ def start_run(run_id=None, experiment_id=None, run_name=None, nested=False):
         if run_name is not None:
             user_specified_tags[MLFLOW_RUN_NAME] = run_name
 
+        if _active_run_stack:
+            user_specified_tags[MLFLOW_ROOT_RUN_ID] = _active_run_stack[0].data.tags[MLFLOW_ROOT_RUN_ID]
```

Comment:
```suggestion
            user_specified_tags[MLFLOW_ROOT_RUN_ID] = _active_run_stack[-1].data.tags[MLFLOW_ROOT_RUN_ID]
```

Given that we append to the active run stack, I think you want the most recent item on the stack, not the first one. Would be good to add a test case that verifies this behavior! (i.e. a test case that would fail if the code were as is, and passes with the code updated to get the most-recent item on the stack)